subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/transform/AttributeMatchingVariantSelector.java

```diff
@@ -111,4 +115,25 @@ private ResolvedArtifactSet doSelect(ResolvedVariantSet producer) {
         }
         throw new NoMatchingVariantSelectionException(producer.asDescribable().getDisplayName(), componentRequested, producer.getVariants(), matcher);
     }
+
+    private List<Pair<ResolvedVariant, ConsumerVariantMatchResult.ConsumerVariant>> tryDisambiguate(List<Pair<ResolvedVariant, ConsumerVariantMatchResult.ConsumerVariant>> candidates) {
+        List<Pair<ResolvedVariant, ConsumerVariantMatchResult.ConsumerVariant>> shorterTransforms = Lists.newArrayListWithExpectedSize(candidates.size());
+
+        for (Pair<ResolvedVariant, ConsumerVariantMatchResult.ConsumerVariant> candidate : candidates) {
+            boolean newCandidate = true;
+            for (int i = 0; i < shorterTransforms.size(); i++) {
```

Comment:
I find this algorithm a little bit hard to read. IIUC, we replace all transformations in the list by the shortest suffix if it is in the same list.

So maybe `shorterTransforms` should be called `shortestTransforms`? WDYT? I would need to think harder to make this algorithm better to read.

Reply:
Will do the rename before merge and see if I can make this simpler after.